### PR TITLE
Issue 277

### LINKED ISSUE #277: Create instruction and/or image for running snaps-hcp @ Cox
We need to be able to bring up our AE @ Cox easily.

### LINKED ISSUE #306: AE: Create a playbook to create AE AMI.
Need the ability to recreate the setup of the ElasticSearch platform on which the new AE is running.

### PR DESCRIPTION
**Pull request by akadam**
Fixes #306 Fixes #277 
#### Questions:
- Have you connected this PR to the issue it resolves? Yes
- Does the documentation need an update? No
- Does this add new dependencies? Yes
- Have you added unit or functional tests for this PR? no

### CODE REVIEW

File: playbooks/tofino/setup_ae_elasticsearch.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Copyright (c) 2019 Cable Television Laboratories, Inc.
```
Comment (spisarski) [inline playbooks/tofino/setup_ae_elasticsearch.yml:1]: Please remove this file

File: playbooks/tofino/setup_nodes-lab_trial.yml [hunk 2]
```diff
@@ -12,7 +12,8 @@
 # limitations under the License.
 ---
 #Import playbook
-- import_playbook: setup_hcp.yml
+#- import_playbook: setup_hcp.yml
```
Comment (spisarski) [inline playbooks/tofino/setup_nodes-lab_trial.yml:15]: Please remove these commented out lines

File: playbooks/tofino/setup_nodes_ae.yml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+---
+#Import playbook
+- import_playbook: setup_ae_centos.yml
+#- import_playbook: setup_ae_elasticsearch.yml
```
Comment (spisarski) [inline playbooks/tofino/setup_nodes_ae.yml:16]: Please remove commented out line.

**Review by spisarski — CHANGES_REQUESTED** [review]
Seeing that CI is using the new image, other than the 3 little cleanup comments, I'm good with this. I did; however, create #320 to address the refactoring we spoke about.

**Comment by spisarski** [comment]
Run Tofino aggregate build

**Comment by spisarski** [comment]
Run Tofino lab trial build

**Comment by spisarski** [comment]
Run Tofino aggregate build

**Comment by spisarski** [comment]
Run Tofino Environment Build

**Review by spisarski — APPROVED** [review]

**Comment by spisarski** [comment]
Run Tofino aggregate build

**Comment by akadam** [comment]
Run Tofino Environment Build

**Comment by akadam** [comment]
Run Tofino lab trial build

**Comment by akadam** [comment]
Run Tofino aggregate build

**Comment by spisarski** [comment]
Run Tofino lab trial build

File: playbooks/tofino/setup_nodes_ae.yml [hunk 4]
```diff
@@ -0,0 +1,15 @@
+# Copyright (c) 2019 Cable Television Laboratories, Inc.
```
Comment (spisarski) [inline playbooks/tofino/setup_nodes_ae.yml:1]: Can you rename this playbook to something like ae-node-buld.yml as the playbook naming will imply building of the environment vs. setting up the AE node for testing. Also, can you move this down one directory called 'ae' like with setup_ae_centos.yml as well?

File: playbooks/tofino/setup_nodes-lab_trial.yml [hunk 5]
```diff
@@ -12,7 +12,6 @@
 # limitations under the License.
 ---
 #Import playbook
```
Comment (spisarski) [inline playbooks/tofino/setup_nodes-lab_trial.yml:14]: How come we are no longer making a call to setup_hcp.yml or is this something that you're going to change in your next PR?

File: playbooks/tofino/setup_hcp.yml [hunk 6]
```diff
@@ -15,12 +15,12 @@
...
   become: yes
   tasks:
-  - name: Install yum dependencies
+  - name: Install setup hcp yum dependencies
     yum:
       name:
         - tcpdump
         - java
-        - logstash
+
```
Comment (spisarski) [inline playbooks/tofino/setup_hcp.yml:23]: Please remove extra line while you are making adjustments

File: playbooks/tofino/setup_ae_centos.yml [hunk 7]
```diff
@@ -0,0 +1,186 @@
+# Copyright (c) 2019 Cable Television Laboratories, Inc.
```
Comment (spisarski) [inline playbooks/tofino/setup_ae_centos.yml:1]: Can you move this file down to a new directory called 'ae'?

**Review by spisarski — COMMENTED** [review]
Please address the comments below and replace the snaps-hcp machine with one that has the Python environment pre-installed with the TPS python requirements.
I've also not seen the lab_trial tests run to completion. I am currently running the tests to see if it is just something strange or an actual issue.

**Comment by spisarski** [comment]
I just fixed the issue where the tofino lab_trial scenario would fail while sniffing for packets on the AE due to the scapy version of 2.4.4 will not work on this new image and needed to downgrade to 2.4.3

**Comment by spisarski** [comment]
Run Tofino Environment Build

**Comment by spisarski** [comment]
Run Tofino aggregate build

**Comment by spisarski** [comment]
Run Tofino lab trial build

File: automation/env-build/aws-security.tf [hunk 8]
```diff
@@ -43,6 +43,20 @@ resource "aws_security_group" "transparent-security-img-sg" {
     protocol = "tcp"
   }
 
+  ingress {
```
Comment (spisarski) [inline automation/env-build/aws-security.tf:46]: Do these ports really need to be opened for building the environment?

File: playbooks/ae/setup_ae_centos.yml [hunk 9]
```diff
@@ -0,0 +1,186 @@
...
+      args:
+        chdir: /home/centos
+
+    - name: Update OpenDistro Elasticsearch Kibana config file
+      become: yes
+      copy:
+        src: ./templates/opendistro_kibana.sh
+        dest: /etc/kibana/kibana.yml
+        force: yes
+
```
Comment (spisarski) [inline playbooks/ae/setup_ae_centos.yml:186]: You should add the installation of the TPS Python dependencies here. All you need to do is copy over requirements.txt and run the equivalent command to "pip3 install -r requirements.txt"

**Review by spisarski — CHANGES_REQUESTED** [review]
It does not look like the new AMI has TPS Python dependencies installed. Please create new AMI with those.

Once you make that change, please let me know your thoughts about not opening port 9200 & 5601 while building the AE environment & what are the plans for setup_hcp.yml seeing that you removed it from the tofino lab_trial scenario.

File: automation/env-build/aws-security.tf (hunk 8 above)
Comment (akadam) [reply automation/env-build/aws-security.tf:46]: Maybe not. Let me test with new AMI and see if isn't causing any issues.I will be re-adding setup_hcp.yml in my next pull.

File: playbooks/ae/setup_ae_centos.yml [hunk 10]
```diff
@@ -0,0 +1,195 @@
...
+
+    - name: Copy TPS requirements.txt
+      copy:
+       src: ../../requirements.txt
```
Comment (spisarski) [inline playbooks/ae/setup_ae_centos.yml:190]: Just a small nit, can we indent two spaces here?

**Review by spisarski — CHANGES_REQUESTED** [review]
After you update the AMI in the automation/p4/tofino/variables.tf and take care of my little nit we should be gtg.

File: playbooks/ae/setup_hcp.yml [hunk 11]
```diff
@@ -15,12 +15,12 @@
...
   become: yes
   tasks:
-  - name: Install yum dependencies
+  - name: Install setup hcp yum dependencies
     yum:
       name:
         - tcpdump
         - java
-        - logstash
+
```
Comment (spisarski) [inline playbooks/ae/setup_hcp.yml:23]: Just a small nit, can you remove this line?

**Comment by spisarski** [comment]
Run Tofino aggregate build

**Comment by spisarski** [comment]
Run Tofino aggregate build

**Review by spisarski — APPROVED** [review]
I will merge after I am able to manually run the CI successfully

**Comment by bo-quan** [comment]
Run Tofino aggregate build

**Comment by bo-quan** [comment]
Run Tofino aggregate build

**Pull request merged**